src/vs/base/node/shell.ts

```diff
@@ -24,6 +25,23 @@ export function getSystemShell(p: platform.Platform, env = process.env as platfo
 	if (platform.isLinux && p === platform.Platform.Mac || platform.isMacintosh && p === platform.Platform.Linux) {
 		return '/bin/bash';
 	}
+
+	return getSystemShellUnixLike(env);
+}
+
+export function getSystemShellSync(p: platform.Platform, env = process.env as platform.IProcessEnvironment): string {
+	if (p === platform.Platform.Windows) {
+		if (platform.isWindows) {
+			return getSystemShellWindowsSync(env);
+		}
+		// Don't detect Windows shell when not on Windows
+		return processes.getWindowsShell(env);
+	}
+	// Only use $SHELL for the current OS
+	if (platform.isLinux && p === platform.Platform.Mac || platform.isMacintosh && p === platform.Platform.Linux) {
+		return '/bin/bash';
+	}
+
 	return getSystemShellUnixLike(env);
```

Comment:
Let's pull this into a function to share code with `getSystemShell`

Reply:
moved into `getSystemShellUnixLike`